spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/connection/TcpMessageMapper.java

```diff
@@ -132,6 +132,10 @@ protected final void addStandardHeaders(TcpConnection connection, AbstractIntegr
 			.setHeader(IpHeaders.IP_ADDRESS, connection.getHostAddress())
 			.setHeader(IpHeaders.REMOTE_PORT, connection.getPort())
 			.setHeader(IpHeaders.CONNECTION_ID, connectionId);
+		SocketInfo socketInfo = connection.getSocketInfo();
+		if (socketInfo != null) {
+			messageBuilder.setHeader(IpHeaders.LOCAL_ADDRESS, socketInfo.getLocalAddress());
```

Comment:
So, have you left `socketInfo` header population as an option for the custom `TcpMessageMapper.supplyCustomHeaders()` implementation ?